File: app/src/main/java/com/afollestad/photoaffix/views/SquareImageView.java
package com.afollestad.photoaffix.views;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.support.v4.content.ContextCompat;
import android.util.AttributeSet;
import android.widget.ImageView;

import com.afollestad.photoaffix.R;

/**
 * @author Aidan Follestad (afollestad)
 */
public class SquareImageView extends ImageView {

    private Paint edgePaint;
    private int borderRadius;

    public SquareImageView(Context context) {
        super(context);
        init();
    }

    public SquareImageView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public SquareImageView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {
        int mAccentColor = ContextCompat.getColor(getContext(), R.color.colorAccent);
        borderRadius = (int) getResources().getDimension(R.dimen.circle_border_radius);
        edgePaint = new Paint();
        edgePaint.setAntiAlias(true);
        edgePaint.setStyle(Paint.Style.STROKE);
        edgePaint.setColor(mAccentColor);
        edgePaint.setStrokeWidth(borderRadius);
    }

    @Override protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        setMeasuredDimension(getMeasuredWidth(), getMeasuredWidth());
    }

    @Override protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        if (isActivated()) {
            int left = borderRadius;
            int top = borderRadius;
            int bottom = getMeasuredHeight() - borderRadius;
            int right = getMeasuredWidth() - borderRadius;
            canvas.drawRect(left, top, right, bottom, edgePaint);
        }
    }
}

File: app/src/main/java/com/afollestad/photoaffix/components/SquareImageView.kt
/*
 * Licensed under Apache-2.0
 *
 * Designed and developed by Aidan Follestad (@afollestad)
 */
package com.afollestad.photoaffix.components

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView
import androidx.core.content.ContextCompat.getColor
import com.afollestad.photoaffix.R

/** @author Aidan Follestad (afollestad) */
class SquareImageView(
  context: Context,
  attrs: AttributeSet? = null
) : AppCompatImageView(context, attrs) {

  private val edgePaint: Paint
  private val borderRadius: Int = context.resources.getDimension(R.dimen.circle_border_radius)
      .toInt()

  init {
    val accentColor = getColor(context, R.color.colorAccent)
    edgePaint = Paint().apply {
      isAntiAlias = true
      style = Paint.Style.STROKE
      color = accentColor
      strokeWidth = borderRadius.toFloat()
    }
  }

  override fun onMeasure(
    widthMeasureSpec: Int,
    heightMeasureSpec: Int
  ) {
    super.onMeasure(widthMeasureSpec, widthMeasureSpec)
  }

  override fun onDraw(canvas: Canvas) {
    super.onDraw(canvas)
    if (isActivated) {
      val left = borderRadius
      val top = borderRadius
      val bottom = measuredHeight - borderRadius
      val right = measuredWidth - borderRadius
      canvas.drawRect(left.toFloat(), top.toFloat(), right.toFloat(), bottom.toFloat(), edgePaint)
    }
  }
}
